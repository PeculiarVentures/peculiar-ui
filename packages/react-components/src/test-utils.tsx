import React from 'react';
import {
  render,
  screen,
  fireEvent,
  waitFor,
  renderHook,
  act,
} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { ThemeProvider } from './styles/theme_provider';

const renderWithWrapper = (
  ui: React.ReactElement,
) => (
  render(
    ui,
    {
      wrapper: ThemeProvider,
    },
  )
);

export {
  userEvent,
  fireEvent,
  renderWithWrapper,
  screen,
  waitFor,
  renderHook,
  act,
};
