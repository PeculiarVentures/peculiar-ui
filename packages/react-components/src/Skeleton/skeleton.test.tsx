import React from 'react';
import { renderWithWrapper as render } from '../test-utils';
import { Skeleton } from './index';

describe('<Skeleton />', () => {
  it('should render with default styles', () => {
    const {
      asFragment,
    } = render(<Skeleton>Inside</Skeleton>);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render with height & width', () => {
    const {
      asFragment,
    } = render(<Skeleton height={30} width={50}>Inside</Skeleton>);

    expect(asFragment()).toMatchSnapshot();
  });

  describe('should render correct variants', () => {
    const variants: React.ComponentProps<typeof Skeleton>['variant'][] = [
      'circle',
      'rect',
      'text',
    ];

    variants.forEach((variant) => {
      it(`variant: ${variant}`, () => {
        const {
          asFragment,
        } = render(
          <Skeleton variant={variant}>Inside</Skeleton>,
        );

        expect(asFragment()).toMatchSnapshot();
      });
    });
  });
});
