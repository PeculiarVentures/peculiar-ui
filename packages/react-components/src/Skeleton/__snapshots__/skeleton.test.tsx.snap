// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Skeleton /> should render correct variants variant: circle 1`] = `
<DocumentFragment>
  @keyframes animation-0 {
  0% {
    opacity: 1;
  }

  50% {
    opacity: 0.4;
  }

  100% {
    opacity: 1;
  }
}

@keyframes animation-0 {
  0% {
    opacity: 1;
  }

  50% {
    opacity: 0.4;
  }

  100% {
    opacity: 1;
  }
}

@keyframes animation-0 {
  0% {
    opacity: 1;
  }

  50% {
    opacity: 0.4;
  }

  100% {
    opacity: 1;
  }
}

.emotion-0 {
  background: var(--pv-color-gray-4);
  display: block;
  height: auto;
  -webkit-animation: animation-0 1.5s ease-in-out 0.5s infinite;
  animation: animation-0 1.5s ease-in-out 0.5s infinite;
  border-radius: 50%;
  max-width: -webkit-fit-content;
  max-width: -moz-fit-content;
  max-width: fit-content;
}

.emotion-0>* {
  visibility: hidden;
}

<div
    class="emotion-0"
  >
    Inside
  </div>
</DocumentFragment>
`;

exports[`<Skeleton /> should render correct variants variant: rect 1`] = `
<DocumentFragment>
  @keyframes animation-0 {
  0% {
    opacity: 1;
  }

  50% {
    opacity: 0.4;
  }

  100% {
    opacity: 1;
  }
}

@keyframes animation-0 {
  0% {
    opacity: 1;
  }

  50% {
    opacity: 0.4;
  }

  100% {
    opacity: 1;
  }
}

@keyframes animation-0 {
  0% {
    opacity: 1;
  }

  50% {
    opacity: 0.4;
  }

  100% {
    opacity: 1;
  }
}

@keyframes animation-0 {
  0% {
    opacity: 1;
  }

  50% {
    opacity: 0.4;
  }

  100% {
    opacity: 1;
  }
}

.emotion-0 {
  background: var(--pv-color-gray-4);
  display: block;
  height: auto;
  -webkit-animation: animation-0 1.5s ease-in-out 0.5s infinite;
  animation: animation-0 1.5s ease-in-out 0.5s infinite;
  max-width: -webkit-fit-content;
  max-width: -moz-fit-content;
  max-width: fit-content;
}

.emotion-0>* {
  visibility: hidden;
}

<div
    class="emotion-0"
  >
    Inside
  </div>
</DocumentFragment>
`;

exports[`<Skeleton /> should render correct variants variant: text 1`] = `
<DocumentFragment>
  @keyframes animation-0 {
  0% {
    opacity: 1;
  }

  50% {
    opacity: 0.4;
  }

  100% {
    opacity: 1;
  }
}

@keyframes animation-0 {
  0% {
    opacity: 1;
  }

  50% {
    opacity: 0.4;
  }

  100% {
    opacity: 1;
  }
}

@keyframes animation-0 {
  0% {
    opacity: 1;
  }

  50% {
    opacity: 0.4;
  }

  100% {
    opacity: 1;
  }
}

@keyframes animation-0 {
  0% {
    opacity: 1;
  }

  50% {
    opacity: 0.4;
  }

  100% {
    opacity: 1;
  }
}

.emotion-0 {
  background: var(--pv-color-gray-4);
  display: block;
  height: auto;
  -webkit-animation: animation-0 1.5s ease-in-out 0.5s infinite;
  animation: animation-0 1.5s ease-in-out 0.5s infinite;
  margin-top: 0;
  margin-bottom: 0;
  transform-origin: 0 55%;
  -webkit-transform: scale(1, 0.60);
  -moz-transform: scale(1, 0.60);
  -ms-transform: scale(1, 0.60);
  transform: scale(1, 0.60);
  border-radius: 4px;
  max-width: -webkit-fit-content;
  max-width: -moz-fit-content;
  max-width: fit-content;
}

.emotion-0:empty:before {
  content: "\\00a0";
}

.emotion-0>* {
  visibility: hidden;
}

<div
    class="emotion-0"
  >
    Inside
  </div>
</DocumentFragment>
`;

exports[`<Skeleton /> should render with default styles 1`] = `
<DocumentFragment>
  @keyframes animation-0 {
  0% {
    opacity: 1;
  }

  50% {
    opacity: 0.4;
  }

  100% {
    opacity: 1;
  }
}

.emotion-0 {
  background: var(--pv-color-gray-4);
  display: block;
  height: auto;
  -webkit-animation: animation-0 1.5s ease-in-out 0.5s infinite;
  animation: animation-0 1.5s ease-in-out 0.5s infinite;
  margin-top: 0;
  margin-bottom: 0;
  transform-origin: 0 55%;
  -webkit-transform: scale(1, 0.60);
  -moz-transform: scale(1, 0.60);
  -ms-transform: scale(1, 0.60);
  transform: scale(1, 0.60);
  border-radius: 4px;
  max-width: -webkit-fit-content;
  max-width: -moz-fit-content;
  max-width: fit-content;
}

.emotion-0:empty:before {
  content: "\\00a0";
}

.emotion-0>* {
  visibility: hidden;
}

<div
    class="emotion-0"
  >
    Inside
  </div>
</DocumentFragment>
`;

exports[`<Skeleton /> should render with height & width 1`] = `
<DocumentFragment>
  @keyframes animation-0 {
  0% {
    opacity: 1;
  }

  50% {
    opacity: 0.4;
  }

  100% {
    opacity: 1;
  }
}

@keyframes animation-0 {
  0% {
    opacity: 1;
  }

  50% {
    opacity: 0.4;
  }

  100% {
    opacity: 1;
  }
}

.emotion-0 {
  background: var(--pv-color-gray-4);
  display: block;
  height: 30px;
  -webkit-animation: animation-0 1.5s ease-in-out 0.5s infinite;
  animation: animation-0 1.5s ease-in-out 0.5s infinite;
  margin-top: 0;
  margin-bottom: 0;
  transform-origin: 0 55%;
  -webkit-transform: scale(1, 0.60);
  -moz-transform: scale(1, 0.60);
  -ms-transform: scale(1, 0.60);
  transform: scale(1, 0.60);
  border-radius: 4px;
  width: 50px;
}

.emotion-0:empty:before {
  content: "\\00a0";
}

.emotion-0>* {
  visibility: hidden;
}

<div
    class="emotion-0"
  >
    Inside
  </div>
</DocumentFragment>
`;
